lib/shared/assert.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * @fileoverview Assertion utilities equivalent to the Node.js node:asserts module.
+ * @author Josh Goldberg
+ */
+
+"use strict";
+
+/**
+ * Throws an error if the input is not truthy.
+ * @param {unknown} value The input that is checked for being truthy.
+ * @param {string} message Message to throw if the input is not truthy.
+ * @returns {void}
+ * @throws {Error}
+ */
+function ok(value, message) {
+    if (!value) {
+        throw new Error(message);
+    }
+}
+
+/**
+ * Throws an error if the input is not strictly equal to expected.
+ * @template T
+ * @param {T} value The input that is checked for being strictly equal.
+ * @param {T} expected The expected value.
+ * @param {string} message Message to throw if the input is not strictly equal.
+ * @returns {void}
+ * @throws {Error}
+ */
+function strictEqual(value, expected, message) {
+    if (value !== expected) {
+        throw new Error(message);
+    }
+}
+
+/**
+ * Throws an error if the input is strictly equal to expected.
+ * @template T
+ * @param {T} value The input that is checked for being strictly equal.
+ * @param {T} expected The expected value.
+ * @param {string} message Message to throw if the input is strictly equal.
+ * @returns {void}
+ * @throws {Error}
```

Comment:
```suggestion
 * @throws {Error}  if the input is strictly equal to the expected value.
```

Reply:
@amareshsm I think you forgot the word "not" in the description

Reply:
```suggestion
 * @throws {Error}  if the input is not strictly equal to the expected value.
```